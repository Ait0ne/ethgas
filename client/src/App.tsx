import React, { useState } from 'react';

import {AppContainer} from './App.styles';
import NotificationForm from './components/NotificationForm/notification-form.component';
import Alert from './components/Alert/alert.component';
import CancelationForm from './components/CancelationForm/cancelation-form.component';

const App:React.FC = () => {
  const [alert, setAlert] = useState({
    shown: false,
    message: '',
    success: true
  })
  const [notificationFormShown, setNotificationFormShow] = useState(true)
  
  const toggleAlert = (message?:string, success?:boolean) => {
      if (message) {
        setAlert({
          shown:true,
          message: message,
          success: success? true : false
        })
      } else {
        setAlert({
          shown: false,
          message: '',
          success: true
        })
      }
  }

  const toggleNotificationForm = () => {
    setNotificationFormShow(c=>!c)
  }

  return (
    <AppContainer>
        <h1>Get notified each time ETH Gas price drops below certain threshold</h1>
        { 
          alert.shown?
          <Alert onClose={toggleAlert} message={alert.message} success={alert.success}/>
          :null
        }
        {
          notificationFormShown?
          <NotificationForm toggleAlert={toggleAlert}/>
          :
          <CancelationForm toggleAlert={toggleAlert} />
        }
        <span onClick={toggleNotificationForm}>{notificationFormShown? 'Want to cancel subscription?': 'Want to subscribe?'}</span>
    </AppContainer>
  );
}

export default App;
