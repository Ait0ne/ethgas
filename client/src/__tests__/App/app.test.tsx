import React from 'react';
import {shallow} from 'enzyme';

import App from '../../App';

describe('app', () => {
    const app = shallow(<App/>)

    it('renders correctly', ()=> {
        expect(app).toMatchSnapshot()
    })

    it('renders notification form', () => {
        expect(app.find('NotificationForm').length).toEqual(1)
    })


    it('renders unsubsribtion text', () => {
        expect(app.find('span').text()).toEqual('Want to cancel subscription?')
    })

    it('shows cancelation form on unsubscription click', () => {
        const unsubscription = app.find('span')
        unsubscription.simulate('click')
        expect(app.find('CancelationForm').length).toEqual(1)
    })

    it('renders subsribtion text', () => {
        expect(app.find('span').text()).toEqual('Want to subscribe?')
    })

    it('doesnt show an alert', () => {
        expect(app.find('Alert').length).toEqual(0)
    })
})