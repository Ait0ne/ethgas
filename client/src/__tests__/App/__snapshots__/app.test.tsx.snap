// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders correctly 1`] = `
<styled.div>
  <h1>
    Get notified each time ETH Gas price drops below certain threshold
  </h1>
  <NotificationForm
    toggleAlert={[Function]}
  />
  <span
    onClick={[Function]}
  >
    Want to cancel subscription?
  </span>
</styled.div>
`;
